packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -144,10 +145,11 @@ export default createRule<[], MessageId>({
                 ]),
               ];
 
-              const stringValue = getStaticStringValue(expression);
-
-              if (stringValue != null) {
-                const escapedValue = stringValue.replace(/([`$\\])/g, '\\$1');
+              if (isLiteral(expression)) {
+                const escapedValue =
+                  typeof expression.value === 'string'
+                    ? expression.raw.slice(1, -1).replace(/([`$])/g, '\\$1')
```

Comment:
(nit, optional) - why are the escaping patterns different for each branch? Maybe worth a comment?

Reply:
Yeah, it does seem a bit confusing

Does https://github.com/typescript-eslint/typescript-eslint/pull/8688/commits/e07e1460765fe2f28fe26fbf6bde967412d84759 clear it up?

Reply:
oof, i think i get it, but i still struggle tbh 😆 It's just really tricky to me to figure out which things are in escaped-land and which things are non-escaped in the code comment there (it's even hard for me to type out the special characters in this comment 😆 ). Is the idea 
raw string &rArr; 
   1. Removing enclosing quotation characters
   1. just add escapes to characters that need escape in template string, but didn't as a string literal (` and $)
   
cooked string &rArr;
   1. escape everything that needs escapes (`, $, and \\)
   
I really think a few English words would be helpful in the code comments (to go along with the examples), since my brain is not big enough to make meaning of all the special characters at a glance 😄